autoscheduler/autoscheduler/settings/base.py

```diff
@@ -13,7 +15,15 @@
 # See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'ogxjva5h%&5c7&e#-2f1&+u5p#zygwffcy!@)k8i37#j_89xe2'
+if _IS_GCP:
+    # For production, load the secret from secret.txt and set it
+    # This is set from the GitHub actions secret DJANGO_SECRET in deploy-workflow.yml
+    _SECRET_PATH = os.path.dirname(__file__) + "/secret.txt"
+    _SECRET_FILE = open(_SECRET_PATH, "r")
+
+    SECRET_KEY = _SECRET_FILE.read()
```

Comment:
```suggestion
    with open(_SECRET_PATH) as _SECRET_FILE:
        SECRET_KEY = _SECRET_FILE.read()[:-1]
```
A couple things that I changed here:
- We should always use context managers when opening files (would've helped here as you forgot to close the file)
- `"r"` is the default mode with open, so no need to use it as an argument
- `file.read()` reads the whole file, including the newline created by `echo`

Reply:
Fixed